utils/log.go

```diff
@@ -0,0 +1,83 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/dgraph-io/badger/v3"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var ErrUnknownLogLevel = errors.New("unknown log level")
+
+type LogLevel uint8
+
+const (
+	DEBUG LogLevel = iota
+	INFO
+	WARN
+	ERROR
+)
+
+func (l LogLevel) String() string {
+	switch l {
+	case DEBUG:
+		return "debug"
+	case INFO:
+		return "info"
+	case WARN:
+		return "warn"
+	case ERROR:
+		return "error"
+	default:
+		return ""
+	}
+}
+
+func (l LogLevel) IsValid() bool {
+	return l.String() != ""
+}
+
+type Logger interface {
+	SimpleLogger
+	badger.Logger
+}
+
+type SimpleLogger interface {
+	Debugw(msg string, keysAndValues ...any)
+	Infow(msg string, keysAndValues ...any)
+	Warnw(msg string, keysAndValues ...any)
+	Errorw(msg string, keysAndValues ...any)
+}
+
+type ZapLogger struct {
+	*zap.SugaredLogger
+}
+
+var _ Logger = (*ZapLogger)(nil)
```

Comment:
Can you please explain what is happening here?

Reply:
Ensures `ZapLogger` fulfills the `Logger` interface. I've seen it in a lot of go codebases, and the Uber style guide [recommends it](https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance). Also makes maintenance a tad easier since gopls will complain until the interface is fulfilled.